src/plugins/explore/public/components/visualizations/rule_repository.ts

```diff
@@ -48,6 +48,7 @@ const oneMetricOneDateRule: VisualizationRule = {
   description: 'Time series visualization for single metric',
   matches: (numerical, categorical, date) =>
     numerical.length === 1 && date.length === 1 && categorical.length === 0,
+  matchIndex: [1, 0, 1],
```

Comment:
Seems there is already `matches`, do we still need matchIndex, or what's the difference?

Reply:
For the visualization empty state component, we need to identify which visualization rules are applicable based on available data columns. The matching logic should use a "greater than or equal to" comparison rather than exact matching, meaning a rule is valid if the available columns meet or exceed the rule's minimum requirements. To support this, we may introduce a matchIndex parameter that indicates the degree of compatibility, making it easier for the field selection component to prioritize and present options to users.